homeassistant/components/roborock/vacuum.py

```diff
@@ -0,0 +1,181 @@
+    12: STATE_ERROR,  # "Error"
+    13: STATE_IDLE,  # "Shutting down"
+    14: STATE_DOCKED,  # "Updating"
+    15: STATE_RETURNING,  # "Docking"
+    16: STATE_CLEANING,  # "Going to target"
+    17: STATE_CLEANING,  # "Zoned cleaning"
+    18: STATE_CLEANING,  # "Segment cleaning"
+    22: STATE_DOCKED,  # "Emptying the bin" on s7+
+    23: STATE_DOCKED,  # "Washing the mop" on s7maxV
+    26: STATE_RETURNING,  # "Going to wash the mop" on s7maxV
+    100: STATE_DOCKED,  # "Charging complete"
+    101: STATE_ERROR,  # "Device offline"
+}
+
+
+ATTR_STATUS = "status"
+ATTR_ERROR = "error"
+
+
+async def async_setup_entry(
+    hass: HomeAssistant,
+    config_entry: ConfigEntry,
+    async_add_entities: AddEntitiesCallback,
+) -> None:
+    """Set up the Roborock sensor."""
+    coordinator: RoborockDataUpdateCoordinator = hass.data[DOMAIN][
+        config_entry.entry_id
+    ]
+    async_add_entities(
+        RoborockVacuum(slugify(device_id), device_info, coordinator)
+        for device_id, device_info in coordinator.api.device_map.items()
+    )
+
+
+class RoborockVacuum(RoborockCoordinatedEntity, StateVacuumEntity):
+    """General Representation of a Roborock vacuum."""
+
+    _attr_icon = "mdi:robot-vacuum"
+    _attr_supported_features = (
+        VacuumEntityFeature.PAUSE
+        | VacuumEntityFeature.STOP
+        | VacuumEntityFeature.RETURN_HOME
+        | VacuumEntityFeature.FAN_SPEED
+        | VacuumEntityFeature.BATTERY
+        | VacuumEntityFeature.STATUS
+        | VacuumEntityFeature.SEND_COMMAND
+        | VacuumEntityFeature.LOCATE
+        | VacuumEntityFeature.CLEAN_SPOT
+        | VacuumEntityFeature.STATE
+        | VacuumEntityFeature.START
+    )
+    _attr_fan_speed_list = list(FAN_SPEED_CODES.values())
+
+    def __init__(
+        self,
+        unique_id: str,
+        device: RoborockDeviceInfo,
+        coordinator: RoborockDataUpdateCoordinator,
+    ) -> None:
+        """Initialize a vacuum."""
+        StateVacuumEntity.__init__(self)
+        RoborockCoordinatedEntity.__init__(self, unique_id, device, coordinator)
+
+    @property
+    def state(self) -> str | None:
+        """Return the status of the vacuum cleaner."""
+        return STATE_CODE_TO_STATE.get(self._device_status.state)
+
+    @property
+    def status(self) -> str | None:
+        """Return the status of the vacuum cleaner."""
+        return self._device_status.status
+
+    @property
+    def battery_level(self) -> int | None:
+        """Return the battery level of the vacuum cleaner."""
+        return self._device_status.battery
+
+    @property
+    def fan_speed(self) -> str | None:
+        """Return the fan speed of the vacuum cleaner."""
+        return self._device_status.fan_power
+
+    @property
+    def error(self) -> str | None:
+        """Get the error str if an error code exists."""
+        return self._device_status.error
+
+    @property
+    def capability_attributes(self) -> dict[str, list[str]]:
```

Comment:
This seems to just be duplicating code in the parent class and is not needed. (Plus `FAN_SPEED` is always present so its more complicated than needs to be)